Composer/packages/client/src/components/AddRemoteSkillModal/SelectIntent.tsx

```diff
@@ -0,0 +1,367 @@
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { Label } from 'office-ui-fabric-react/lib/Label';
+import { LuEditor } from '@bfc/code-editor';
+import { ScrollablePane, ScrollbarVisibility } from 'office-ui-fabric-react/lib/ScrollablePane';
+import { LuFile, LuIntentSection, SDKKinds } from '@bfc/shared';
+import { useRecoilValue } from 'recoil';
+
+import TelemetryClient from '../../telemetry/TelemetryClient';
+import { selectIntentDialog, enableOrchestratorDialog } from '../../constants';
+import httpClient from '../../utils/httpUtil';
+import luWorker from '../../recoilModel/parsers/luWorker';
+import { localeState, dispatcherState } from '../../recoilModel';
+import { recognizersSelectorFamily } from '../../recoilModel/selectors/recognizers';
+
+import { Orchestractor } from './Orchestractor';
+import { getLuDiagnostics } from './helper';
+
+const detailListContainer = css`
+  width: 100%;
+  height: 300px;
+  position: relative;
+  overflow: hidden;
+  flex-grow: 1;
+  border: 1px solid E5E5E5;
+`;
+
+interface SelectIntentProps {
+  manifest: {
+    dispatchModels: {
+      intents: Array<string> | object;
+      languages: object;
+    };
+    [key: string]: any;
+  };
+  languages: string[];
+  projectId: string;
+  luFeatures: any;
+  rootLuFiles: LuFile[];
+  dialogId: string;
+  onSubmit: (event, content: string, enable: boolean) => void;
+  onDismiss: () => void;
+  setTitle: (value) => void;
+  onBack: () => void;
+}
+
+const columns = [
+  {
+    key: 'Name',
+    name: 'Name',
+    minWidth: 300,
+    isResizable: false,
+    onRender: (item: string) => {
+      return <Fragment>{item}</Fragment>;
+    },
+  },
+];
+
+const getRemoteLuFiles = async (skillLanguages: object, composerLangeages: string[], setWarningMsg) => {
+  const luFilePromise: Promise<any>[] = [];
+  try {
+    for (const [key, value] of Object.entries(skillLanguages)) {
+      if (composerLangeages.includes(key)) {
+        value.map((item) => {
+          // get lu file
+          luFilePromise.push(
+            httpClient
+              .get(`/utilities/retrieveRemoteFile`, {
+                params: {
+                  url: item.url,
+                },
+              })
+              .catch((err) => {
+                console.error(err);
+                setWarningMsg('get remote file fail');
+              })
+          );
+        });
+      }
+    }
+    const responses = await Promise.all(luFilePromise);
+    const files: { id: string; content: string }[] = responses.map((response) => {
+      return response.data;
+    });
+    return files;
+  } catch (e) {
+    console.log(e);
+  }
+};
+
+const getParsedLuFiles = async (files: { id: string; content: string }[], luFeatures, lufiles) => {
+  const promises = files?.map((item) => {
+    return luWorker.parse(item.id, item.content, luFeatures, lufiles) as Promise<LuFile>;
+  });
+  const luFiles: LuFile[] = await Promise.all(promises);

```

Comment:
you have shadowed variable luFiles, use another name for this or change the arg variable.

Reply:
add types to args.